sopel/plugins/callables.py

```diff
@@ -0,0 +1,586 @@
+"""Plugin objects' definition.
+
+.. versionadded:: 8.1
+
+The core of a plugin consists of several objects, either plugin callables or
+plugin jobs. These objects are represented by instances of
+:class:`PluginCallable` and :class:`PluginJob` respectively.
+
+.. seealso::
+
+    The :mod:`sopel.plugin`'s decorator create these objects directly.
+
+.. important::
+
+    This is all relatively new. Its usage and documentation is for Sopel core
+    development and advanced developers. It is subject to rapid changes
```

Comment:
We might consider dropping "and advanced developers" from this disclaimer in any new core stuff that's not intended as public API, since ideally plugin devs won't touch it no matter how "advanced" they think they are.

This is a public service announcement from the Sopel Crimes™ Stoppers Unit.

Reply:
Done!